src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+
```

Comment:
Maybe it would be a good time to start a `src/compat` folder?

Reply:
Good idea, I will make it.

Reply:
I have made an `/ext` folder as suggested in Julia docs: https://docs.julialang.org/en/v1.10-dev/manual/code-loading/#man-extensions .